alternator/executor.cc

```diff
@@ -2300,22 +2300,22 @@ static std::optional<attrs_to_get> calculate_attrs_to_get(const rjson::value& re
  * as before.
  */ 
 void executor::describe_single_item(const cql3::selection::selection& selection,
-    const std::vector<bytes_opt>& result_row,
+    const std::vector<managed_bytes_opt>& result_row,
```

Comment:
I don't understand if this change is part of the **goal** of this patch, to avoid contiguous allocates when there are large cells (?), or is it a side-effect? Because if it's a side-effect it's unfortunate because this is a performance-sensitive area of the code, part of the Alternator read code path.

Reply:
It 's a side effect, but a desirable one. Of course we don't want to regress alternator.

Reply:
I thought this wasn't a hot path due to the name.

Reply:
Maybe the function `describe_single_item` is confusingly named given the meaning of "describe" in CQL. It converts the query result from Scylla's internal structures to a JSON that needs to be returned for this result, so it happens for every read request (and write requests with read-before-write). It's not a "super hot" path - it only happens once per read row and not many times in a loop, but it's not a rare path either.

It seems you are suggesting that using this query::result API to do the query was a mistake because only non-performance-sensitive code should use it, so Alternator shouldn't have used it. What should I have used instead?

Note that I'm not asking just about the implications of your new managed_bytes changes, but the general statement that this is an inefficient query API by design :-(